src/lib/shell/commands/Dns.cpp

```diff
@@ -39,40 +40,38 @@ Shell::Engine sShellDnsBrowseSubcommands;
 Shell::Engine sShellDnsSubcommands;
 Dnssd::ResolverProxy sResolverProxy;
 
-class DnsShellResolverDelegate : public Dnssd::OperationalResolveDelegate, public Dnssd::CommissioningResolveDelegate
+class DnsShellResolverDelegate : public Dnssd::CommissioningResolveDelegate, public AddressResolve::NodeListener
 {
 public:
-    void OnOperationalNodeResolved(const Dnssd::ResolvedNodeData & nodeData) override
+    DnsShellResolverDelegate() { mSelfHandle.SetListener(this); }
+
+    void OnNodeAddressResolved(const PeerId & peerId, const AddressResolve::ResolveResult & result) override
     {
-        sResolverProxy.NodeIdResolutionNoLongerNeeded(nodeData.operationalData.peerId);
         streamer_printf(streamer_get(), "DNS resolve for " ChipLogFormatX64 "-" ChipLogFormatX64 " succeeded:\r\n",
-                        ChipLogValueX64(nodeData.operationalData.peerId.GetCompressedFabricId()),
-                        ChipLogValueX64(nodeData.operationalData.peerId.GetNodeId()));
-        streamer_printf(streamer_get(), "   Hostname: %s\r\n", nodeData.resolutionData.hostName);
-        for (size_t i = 0; i < nodeData.resolutionData.numIPs; ++i)
-        {
-            streamer_printf(streamer_get(), "   IP address: %s\r\n", nodeData.resolutionData.ipAddress[i].ToString(ipAddressBuf));
-        }
-        streamer_printf(streamer_get(), "   Port: %u\r\n", nodeData.resolutionData.port);
-
-        auto retryInterval = nodeData.resolutionData.GetMrpRetryIntervalIdle();
-
-        if (retryInterval.HasValue())
-            streamer_printf(streamer_get(), "   MRP retry interval (idle): %" PRIu32 "ms\r\n", retryInterval.Value());
-
-        retryInterval = nodeData.resolutionData.GetMrpRetryIntervalActive();
-
-        if (retryInterval.HasValue())
-            streamer_printf(streamer_get(), "   MRP retry interval (active): %" PRIu32 "ms\r\n", retryInterval.Value());
-
-        streamer_printf(streamer_get(), "   Supports TCP: %s\r\n", nodeData.resolutionData.supportsTcp ? "yes" : "no");
+                        ChipLogValueX64(peerId.GetCompressedFabricId()), ChipLogValueX64(peerId.GetNodeId()));
+
+        char addr_string[Transport::PeerAddress::kMaxToStringSize];
+        result.address.ToString(addr_string);
+
+        streamer_printf(streamer_get(), "Resolve completed: %s\r\n", addr_string);
```

Comment:
You should be able to log all the addresses (and in fact all the ports, etc, depending on what responses look like) if this code also does:
```
AddressResolve::Resolver::Instance().TryNextResult(Handle());
```
or so after logging this response.  At least for the backends that usefully provide more than one address.